Controllers/ImageController.cs

```diff
@@ -0,0 +1,336 @@
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost("generateImagesZip")]
+        public async Task<ActionResult> Post([FromForm] ImageFormData Form)
+        {
+            var zipId = Guid.NewGuid();
+
+            try
+            {
+                using (var args = ImageGenerationModel.FromFormData(HttpContext.Request.Form, HttpContext.Request.Form.Files))
+                {
+                    // Punt if we have invalid arguments.
+                    if (!string.IsNullOrEmpty(args.ErrorMessage))
+                    {
+                        return new ObjectResult(args.ErrorMessage) { StatusCode = (int?)HttpStatusCode.BadRequest };
+                    }
+
+                    var profiles = GetProfilesFromPlatforms(args.Platforms);
+                    var imageStreams = new List<Stream>(profiles.Count);
+
+                    using (var zip = ZipFile.Open(CreateFilePathFromId(zipId), ZipArchiveMode.Create))
+                    {
+                        var iconObject = new IconRootObject();
+                        foreach (var profile in profiles)
+                        {
+                            var stream = CreateImageStream(args, profile);
+                            imageStreams.Add(stream);
+                            var fmt = string.IsNullOrEmpty(profile.Format) ? "png" : profile.Format;
+                            var iconEntry = zip.CreateEntry(profile.Name + "." + fmt, CompressionLevel.Fastest);
+                            var iconStream = iconEntry.Open();
+                            await stream.CopyToAsync(iconStream);
+                            iconStream.Close();
+                            stream.Close();
+        
+                            iconObject.icons.Add(new IconObject(profile.Folder + profile.Name + "." + fmt, profile.Width + "x" + profile.Height));
+                        }
+
+                        var iconStr = JsonConvert.SerializeObject(iconObject, Formatting.Indented);
+
+                        using (StreamWriter writer = new StreamWriter(zip.CreateEntry("icons.json", CompressionLevel.Optimal).Open()))
+                        {
+                            writer.Write(iconStr);
+                        }
+
+                        string zipFilePath = CreateFilePathFromId(zipId);
+                        imageStreams.ForEach(s => s.Dispose());
+                    }
+                }
+            }
+            catch (OutOfMemoryException ex)
+            {
+                return new ObjectResult(ex.ToString()) { StatusCode = (int?)HttpStatusCode.UnsupportedMediaType };
+            }
+            catch (Exception ex)
+            {
+                return new ObjectResult(ex.ToString()) { StatusCode = (int?)HttpStatusCode.InternalServerError };
+            }
+
+            // Send back a route to download the zip file.
+            var url = Url.RouteUrl("getImagesZipById", new { id = zipId.ToString() });
```

Comment:
Seems like route could be a good constant since it needs to be reused here.

Reply:
Fixed